### PR TITLE
[WIP] New Architecture: mobile-wallet-adapter-protocol

### PR DESCRIPTION
**Pull request by Michaelsulistio**
Backwards compatible Turbo Module for `mobile-wallet-adapter-protocol`.

## Test plan
For each test case, we test `newArchEnabled` flag both `true/false` and verify MWA works.

### Test cases
1. Verify working integration for local Example App ✅ 
2. Publish `beta` version on NPM and verify it works in new RN project (test multiple versions <= 0.76.2) ❌ 
3. Publish `beta` version on NPM and verify it works in new Expo project (test multiple versions <=Expo SDK 52) ❌ 

Don't merge until testing is complete

### CODE REVIEW

**Comment by Michaelsulistio** [comment]
Local Import Testing:

For these test, I'm importing `mobile-wallet-adapter-protocol` and `mobile-wallet-adapter-protocol-web3js` locally my local repo with newArch changes

React Native Scaffold App
- Local Import, react-native: 0.71.4, newArchEnabled=false ✅ 
- Local Import, react-native: 0.71.4, newArchEnabled=true ✅ 

Expo Template App
- Local import, react-native: 0.76.3, Expo SDK 52, newArchEnabled=true ✅ 
- Local import, react-native: 0.76.3, Expo SDK 52, newArchEnabled=false ✅

**Review by Funkatronics — APPROVED** [review]

**Comment by Funkatronics** [comment]
RN Wallet build is failing on CI for an unknown reason. no error is reported. The example app is building locally for me on this branch so we are okay to merge this I will  monitor main after merge to see if the build error persists.

**Comment by timmyjose** [comment]
Does this mean that React Native New Architecture is now supported?

**Pull request merged**
